File: average.cpp
#include <iostream>
using namespace std;
int main()
{
    int i;
    float k = 0;
    float c = 0;
    while (i != 0)
    {
        cin >> i;
        if (i != 0)
        {
            c += i;
            k += 1;
        }  
    }  
    cout << "average = " << c/k;
    string b;
    cin >> b;
}

File: average.java
import java.util.Scanner;
public class average {
    public static void main(String[] args)
    {
        boolean b = false;
        float k = 0;
        float c = 0;
        while (b != true)
        {
            Scanner scan = new Scanner(System.in);
            int i = scan.nextInt();
            if (i != 0)
            {
                c += i;
                k += 1;
            }
            else {
                break;
            }
        } 
        System.out.println(c/k); 
    }
}